src/gsSolver/gsSimplePreconditioners.h

```diff
@@ -317,11 +351,32 @@ class gsGaussSeidelOp : public gsSteppableOperator<typename MatrixType::Scalar>
         }
     }
 
+    void stepT(const gsMatrix<T> & rhs, gsMatrix<T> & x) const
+    {
+        if (ordering == gsGaussSeidel::forward )
+            reverseGaussSeidelSweep<T>(m_expr,x,rhs);
```

Comment:
Note that calling the free function can be highly inefficient here (temp copy to sparsematrix)

Reply:
I think this should be fixed by changing the type in the corresponding functions to gsSparseMatrix<T>::Nested

Further improvements should/could be considered seperately of this mg merge.